rules/store/store.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package rules
+
+import "github.com/m3db/m3metrics/rules"
+
+// Store performs read/write operations for rules and namespaces.
+type Store interface {
```

Comment:
Would it make sense to keep the definition of the `Store` interface in the rules directory and just move the concrete implementation into the `store` subdirectory? That would allow us to avoid redundant names (e.g. `store.Store`) while also keep all the actual implementation details together in a sub package.

Reply:
Hm yeah I did it that way initially but ended up moving them into their own package. Feels a bit cleaner now all store-related logic is in `store` package and all validator-related logic is in `validator` package. I could move them out into the `rules` package if you feel strongly.

Reply:
I don't feel that strongly. If you feel this approach is cleaner I'm fine with it.

Reply:
Actually it does look better after moving the interface definitions to `rules/`. Will do.